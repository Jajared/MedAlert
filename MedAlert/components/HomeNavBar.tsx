import { StyleSheet, Text, View, TouchableOpacity, Image } from "react-native";
import { AntDesign } from "@expo/vector-icons";

export default function HomeNavBar({ navigation, userName, hasReminders }) {
  const homeIcon = require("../assets/home-icon.png");
  return (
    <View style={styles.container} testID="HomeNavBar">
      <TouchableOpacity onPress={() => navigation.navigate("Add Medication Type")} style={styles.addButton} testID="AddButton">
        <AntDesign name="plus" size={25} color="black" />
      </TouchableOpacity>
      <Text style={styles.userSection}>Hey {userName}! ðŸ˜ƒ</Text>
      <View style={styles.homeContainer}>
        <View style={styles.upcomingReminderSection}>{hasReminders ? <Text style={styles.upcomingRemindersText}>You have upcoming reminders for the day.</Text> : <Text style={styles.upcomingRemindersText}>You have no reminders for the day.</Text>}</View>
        <Image source={homeIcon} style={styles.homeIcon} />
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "column",
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
  userSection: {
    flex: 1,
    width: "90%",
    fontSize: 20,
    fontWeight: "bold",
  },
  upcomingRemindersText: {
    fontSize: 18,
    fontWeight: "400",
  },
  addButton: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "flex-end",
    paddingRight: 30,
    marginTop: 10,
    width: "100%",
  },
  homeContainer: {
    flex: 3,
    backgroundColor: "#FFD9C0",
    flexDirection: "row",
    borderColor: "#FFD9C0",
    borderWidth: 10,
    borderRadius: 12,
    width: "95%",
    padding: 10,
  },
  upcomingReminderSection: {
    flex: 3,
    alignItems: "center",
    justifyContent: "center",
  },
  homeIcon: {
    flex: 2,
    width: "100%",
    height: "100%",
  },
});
