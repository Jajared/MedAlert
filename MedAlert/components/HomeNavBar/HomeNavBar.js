import { StyleSheet, Text, View, TouchableOpacity } from "react-native";
import { AntDesign } from "@expo/vector-icons";

export default function HomeNavBar({ props, navigation, userName }) {
  return (
    <View style={styles.container}>
      <TouchableOpacity onPress={() => navigation.navigate("Add Medication Type")} style={styles.addButton}>
        <AntDesign name="plus" size={25} color="black" />
      </TouchableOpacity>
      <Text style={styles.userSection}>Hey {userName}!</Text>
      <Text style={styles.daySection}>{new Intl.DateTimeFormat("en-US", { weekday: "long" }).format(new Date())}</Text>
      <Text style={styles.upcomingRemindersSection}>Upcoming Reminders </Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "column",
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
  userSection: {
    flex: 1,
    width: "90%",
    fontSize: 20,
    fontWeight: "300",
  },
  upcomingRemindersSection: {
    flex: 1,
    fontSize: 25,
    fontWeight: "bold",
    width: "90%",
  },
  daySection: {
    flex: 1,
    fontSize: 30,
    fontWeight: "bold",
    width: "90%",
  },
  addButton: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "flex-end",
    paddingRight: 30,
    width: "100%",
  },
});
