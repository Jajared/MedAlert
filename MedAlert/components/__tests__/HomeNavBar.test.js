import { render, screen, fireEvent } from "@testing-library/react-native";
import HomeNavBar from "../HomeNavBar";
import { expect, jest, test } from "@jest/globals";

jest.mock("expo-font");
jest.mock("expo-asset");

const navigationObject = {
  navigate: jest.fn(),
  goBack: jest.fn(),
};

describe("HomeNavBar", () => {
  // Check if renders correctly
  it("renders correctly", () => {
    const tree = render(<HomeNavBar navigation={navigationObject} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  // Check number of children
  it("has 3 children", () => {
    render(<HomeNavBar navigation={navigationObject} />);
    expect(screen.getByTestId("HomeNavBar").children.length).toBe(3);
  });
  // Check if add button works
  it("Successfully navigate to add medication", () => {
    render(<HomeNavBar navigation={navigationObject} />);
    fireEvent.press(screen.getByTestId("AddButton"));
    expect(navigationObject.navigate).toBeCalledWith("Add Medication Type");
  });
});
