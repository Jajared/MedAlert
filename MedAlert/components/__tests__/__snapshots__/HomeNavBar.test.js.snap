// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeNavBar renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
  testID="HomeNavBar"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "marginTop": 10,
        "opacity": 1,
        "paddingRight": 30,
        "width": "100%",
      }
    }
    testID="AddButton"
  >
    <Text />
  </View>
  <Text
    style={
      {
        "flex": 1,
        "fontSize": 20,
        "fontWeight": "bold",
        "width": "90%",
      }
    }
  >
    Hey 
    ! ðŸ˜ƒ
  </Text>
  <View
    style={
      {
        "backgroundColor": "#FFD9C0",
        "borderColor": "#FFD9C0",
        "borderRadius": 12,
        "borderWidth": 10,
        "flex": 3,
        "flexDirection": "row",
        "padding": 10,
        "width": "95%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 3,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "400",
          }
        }
      >
        You have no reminders for the day.
      </Text>
    </View>
    <Image
      source={1}
      style={
        {
          "flex": 2,
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;
